TeslaLib/TeslaClient.cs

```diff
@@ -113,7 +113,81 @@ public TeslaClient(string email, string teslaClientId, string teslaClientSecret,
 
             TeslaAuthHelper = authHelper ?? new TeslaAuthHelper(FlexChargingUserAgent, region);
         }
-        public IOAuthTokenStore OAuthTokenStore { get; set; }
+        public IOAuthTokenStore OAuthTokenStore { get; set; } 
+        public async Task LoginUsingTokenStoreAsync(LoginToken loginToken, bool forceRefreshOlderThanToday = false)
+        {
+            if (string.IsNullOrWhiteSpace(Email))
+            {
+                throw new ArgumentNullException(nameof(Email));
+            }
+            var token = await TokenStoreForThisInstance.GetTokenAsync(Email);
+
+            if (token == null)
+            {
+                await TokenStoreForThisInstance.AddTokenAsync(Email, loginToken);
+
+                token = await TokenStoreForThisInstance.GetTokenAsync(Email);
+            }
+            else
+            {
+                await TokenStoreForThisInstance.UpdateTokenAsync(Email, loginToken);

```

Comment:
We have two cases.  
1) The loginToken parameter is the same as the token in the store.  This will be very common.
2) Another process updated the login token in the store, and we might be trying to log in with an obsolete one where the refresh token no longer works.  This may be doing exactly the wrong thing in this case.  
The key thing is we need code outside this project to be regularly loading the token from the token store before calling here, at least every hour.

Reply:
Thanks! Done